### PR TITLE
Simple DoFnRunner optimizatons.

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @silviulica

File: sdks/python/apache_beam/runners/common.py [hunk 1]
```diff
@@ -95,9 +98,8 @@ def finish(self):
 
   def process(self, element):
     try:
-      with self.logger.PerThreadLoggingContext(step_name=self.step_name):
```
Comment (bjchambers) [inline sdks/python/apache_beam/runners/common.py]: I think removing this regresses functionality of the Python SDK relative to Java. We currently ensure that any log message includes the current step in its context to simplify querying, etc.

Is there a plan to re-enable this functionality?

**Comment by silviulica** [comment]
I agree with both of you :-) Ben I think Robert is trying to establish a good performance baseline that we will want to keep for the foreseeable future. We should log a bug that this functionality needs to go back.

**Comment by silviulica** [comment]
LGTM

File: sdks/python/apache_beam/runners/common.py (hunk 1 above)
Comment (robertwb) [inline sdks/python/apache_beam/runners/common.py]: Fair point. PTAL. (Worker side code coming separately, but requires this awkward boundary.)

**Comment by dhalperi** [comment]
Close manually at your convenience

**Pull request closed without merging**